packages/openneuro-app/src/scripts/errors/errorBoundary.jsx

```diff
@@ -28,79 +101,61 @@ class ErrorBoundary extends React.Component {
       supportModal: false,
       error: props.error,
       eventId: null,
+      message: props.error ? props.error.message : '',
     }
   }
-
   static getDerivedStateFromError(error) {
-    return getDerivedStateFromErrorOnCondition(
-      error,
-      // error boundary should always be triggered in general case
-      () => true,
-    )
+    return getDerivedStateFromErrorOnCondition(error, () => true)
   }
-
   componentDidCatch(error) {
+    let message = this.state.message
+    if (!this.props.dataset) {
+      message = 'This dataset does not exist.'
+    } else if (this.props.dataset && this.props.dataset.snapshots.length < 1) {
```

Comment:
Message text is good, but I think users should be pointed to the freshdesk modal on this condition.

Reply:
'This dataset has no associated snapshots.'